src/main/java/org/apache/jackrabbit/filevault/maven/packaging/impl/ImportPackageBuilder.java

```diff
@@ -493,24 +569,45 @@ private BundleInfo(Artifact artifact) throws IOException {
                     }
                     String path = e.getName();
                     if (path.endsWith(".class")) {
-                        // skip internal / impl
-                        if (path.contains("/impl/") || path.contains("/internal/")) {
-                            continue;
-                        }
-                        path = StringUtils.chomp(path, "/");
-                        if (path.charAt(0) == '/') {
-                            path = path.substring(1);
-                        }
-                        String packageName = path.replaceAll("/", ".");
-                        packageVersions.put(packageName, "");
+                        fileNames.add(path);
                     }
                 }
             }
-            jar.close();
+
+            return fileNames;
         }
+    }
 
-        public String getId() {
-            return id;
+    /**
+     * Represents an {@link Artifact} dependency to another module of a multi-module setup
+     * which may point to the target/classes directory of the module rather than a jar file.
+     */
+    private static class DirectoryDependency implements Dependency {
+        private final File directory;
+
+        private DirectoryDependency(File directory) {
+            this.directory = directory;
+        }
+
+        public Manifest getManifest() throws IOException {
+            File manifest = new File(this.directory, "META-INF/MANIFEST.MF");
+            if (!manifest.exists()) {
+                return null;
+            }
+            try (InputStream in = new FileInputStream(manifest)) {
+                return new Manifest(in);
+            }
+        }
+
+        @Override
+        public Collection<String> getClassFiles() throws IOException {
+            Collection<File> files = listFiles(this.directory, new String[]{"class"}, true);
+            Collection<String> fileNames = new ArrayList<>(files.size());
+            for (File file : files) {
+                String platformSpecificFilePath = getPath(file.getCanonicalPath()) + "/" + file.getName();
```

Comment:
is this correct to use the cannonical path? shouldn't it be relative to the `classes` directory?